packages/playback-core/src/autoplay.ts

```diff
@@ -110,12 +110,15 @@ export const setupAutoplay = (
           if (hls) {
             hls.once(Hls.Events.LEVEL_UPDATED, handleSeek);
           } else if (mediaEl.seekable.end(0) === Infinity) {
-            const interval = setInterval(() => {
-              if (mediaEl.seekable.end(0) !== Infinity) {
-                clearInterval(interval);
-                handleSeek();
-              }
-            }, 100);
+            mediaEl.addEventListener(
+              'canplay',
+              () => {
+                // we need a setTimeout because seeking immediately on canplay
+                // can cause the player to stop get into a non-playing state
+                setTimeout(handleSeek, 100);
```

Comment:
100ms seems to be the best and making the player not freeze after the seek, though, it isn't ideal either.

Reply:
blocking: Is 100ms a "magic number"? Is deferring until the next event loop/frame a la `Promise.resolve().then(...)` or `setTimeout(...,0)` or `requestAnimationFrame(...)` not sufficient? If it is a necessary magic number, how confident are you that this will reliably work? Have you tested across envs?

Reply:
Yeah, it's a bit of a magic number playing around in desktop safari. `setTimeout(0)` is *not* sufficient.
I'm not positive that this change is inherently better than what we had previously, unfortunately. But let me double check how it compares to the setInterval and how it works on ipad.